memory.js

```diff
@@ -0,0 +1,116 @@
+(function() {
+
+    var prevSquareVal,
+        currentSquareVal,
+        clickCount = 0;
+
+    var Board = function(size) {
+        this.size = size;
+    };
+
+    Board.prototype.createSquares = function() {
+        var table = document.createElement("table");
+        for (var i = 0; i < this.size; i++) {
+            var newRow = table.insertRow();
+            for (var j = 0; j < this.size; j++) {
+                var newCell = newRow.insertCell();
+                var span = document.createElement("span");
+                newCell.appendChild(span);
+                span.className = "text";
+            }
+            table.appendChild(newRow);
+        }
+        document.body.appendChild(table);
+        this.checkMatches();
+    };
+
+    Board.prototype.createDupArrays = function() {
+        var spans = document.querySelectorAll("span");
+        var spanList = Array.prototype.slice.call(spans);
+        var spanLengthHalved = spanList.length / 2;
+
+        this.numbers = [];
+        this.dupNumbers = [];
+
+        for (var i = 1; i <= spanLengthHalved; i++) {
+            this.numbers.push(i);
+        }
```

Comment:
You might consider moving this to a simple method that creates arrays of a specified length, prepopulated with numbers `1-n`. Then `dupNumbers` could just be another call to that method.